### PR TITLE
add dbt deps logging example

### PR DESCRIPTION
**Pull request by sungchun12**
## Description & motivation

Illustrates to the developer how dbt deps version logging works.

## To-do before merge

- [x] Ensure PR [#3759](https://github.com/dbt-labs/dbt/pull/3759)  is merged

## Pre-release docs
Is this change related to an unreleased version of dbt?
- [x] Yes: please
    - update the base branch to `next`
    - add Changelog components: `<Changelog>[New/Changed] in v0.x.0</Changelog>`
    - add links to the "New and changed documentation" section of the latest [Migration Guide](../website/docs/docs/guides/migration-guide)
- [ ] No: please ensure the base branch is `current`
- [ ] Unsure: we'll let you know!

### CODE REVIEW

**Review by jtcohen6 — COMMENTED** [review]
Very cool, thanks for opening the PR @sungchun12!

What do you think about including this sample output in the page for [`deps`](https://deploy-preview-798--docs-getdbt-com.netlify.app/reference/commands/deps) instead? I don't have a super strong feeling here, just thinking that the page on Packages is already quite dense with configuration options, and the page on `deps` is quite sparse by comparison.

**Comment by sungchun12** [comment]
@jtcohen6 I agree with your suggestion! It gives the `deps` docs page a nice UX to understand what behavior it should produce upfront.

File: website/docs/docs/guides/migration-guide/upgrading-to-0-21-0.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ dbt v0.21.0-b2 is currently available as a prerelease. If you have questions or
 - [Commands: `source`](commands/source): Renamed to `dbt source freshness`, updated selection logic.
 - [Environment variables](env_var): Add a log-scrubbing prefix, `DBT_ENV_SECRET_`
 - [Selection methods](node-selection/methods) and [state comparison caveats](state-comparison-caveats): Add `state:modified` subselectors, and reflect that it now includes changes to upstream macros.
+- [Packages](package-management): Add `dbt deps` logging for outdated packages
```
Comment (jtcohen6) [inline website/docs/docs/guides/migration-guide/upgrading-to-0-21-0.md]:
```suggestion
- [`deps`](commands/deps): Add `dbt deps` logging for outdated packages
```

**Review by jtcohen6 — APPROVED** [review]
lgtm! thanks for the excellent follow-through @sungchun12 :)

**Pull request merged**